### PR TITLE
Fixes SMS not sending for country codes not supporting alphanumeric sender IDs

### LINKED ISSUE #5943: [CAL-574] SMS reminder not working for workflows
Found a bug? Please fill out the sections below. 👍

### Issue Summary

Reported by customer and tested myself, send SMS reminder to a specific number doesn't work in testing or in application. 

### Technical details

<sub>[CAL-574](https://linear.app/calcom/issue/CAL-574/sms-reminder-not-working-for-workflows)</sub>

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

Not all countries support alphanumeric sender IDs for SMS sending. Here is the list of countries supporting/not supporting it: [Supported Countries](https://support.twilio.com/hc/en-us/articles/223133767-International-support-for-Alphanumeric-Sender-ID
)

This caused that SMS sending didn't work for some countries. With this PR we now check if the country code supports alphanumeric sender Ids and then won't use them if it's not supported.

Fixes #5943

**Environment**: Staging(main branch)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- Test workflow SMS action with a phone number that does not support alphanumeric sender IDs
- Test workflow SMS action with a phone number that does support alphanumeric sender IDs

### CODE REVIEW

File: packages/features/ee/workflows/lib/aphanumericSenderIdSupport.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
+export const noAlphanumericSenderIdSupport = [
```
Comment (CarinaWolli) [inline packages/features/ee/workflows/lib/aphanumericSenderIdSupport.ts:1]: List of country codes that don't support alphanumeric sender IDs: 
[Supported Countries](https://support.twilio.com/hc/en-us/articles/223133767-International-support-for-Alphanumeric-Sender-ID)
Country codes from here: [Country Codes](https://countrycode.org/)

File: packages/features/ee/workflows/api/scheduleSMSReminders.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@ import dayjs from "@calcom/dayjs";
 import { defaultHandler } from "@calcom/lib/server";
 import prisma from "@calcom/prisma";
 
+import { noAlphanumericSenderIdSupport } from "../lib/aphanumericSenderIdSupport";
```
Comment (baileypumfleet) [inline packages/features/ee/workflows/api/scheduleSMSReminders.ts:9]:
```suggestion
import { noAlphanumericSenderIdSupport } from "../lib/alphanumericSenderIdSupport";
```
Found a misspelling in this import

File: packages/features/ee/workflows/api/scheduleSMSReminders.ts [hunk 3]
```diff
@@ -72,6 +73,11 @@ async function handler(req: NextApiRequest, res: NextApiResponse) {
           ? reminder.booking?.attendees[0].timeZone
           : reminder.booking?.user?.timeZone;
 
+      const isAlphanumericSenderIdSupported = !noAlphanumericSenderIdSupport.find(
+        (code) => code === sendTo?.substring(0, code.length)
+      );
+      const senderID = isAlphanumericSenderIdSupported ? reminder.workflowStep.sender || "Cal" : "";
```
Comment (baileypumfleet) [inline packages/features/ee/workflows/api/scheduleSMSReminders.ts:79]:
```suggestion
      const senderID = isAlphanumericSenderIdSupported ? reminder.workflowStep.sender || "Cal.com" : "";
```
Can we use Cal.com as the sender ID instead? It's a little more clear than just Cal

File: packages/features/ee/workflows/components/WorkflowStepContainer.tsx [hunk 4]
```diff
@@ -596,6 +596,7 @@ export default function WorkflowStepContainer(props: WorkflowStepProps) {
                 reminderBody: reminderBody || "",
                 template: step.template,
                 sendTo: step.sendTo || "",
+                sender: step.sender || "Cal",
```
Comment (baileypumfleet) [inline packages/features/ee/workflows/components/WorkflowStepContainer.tsx:599]: Here too

File: packages/features/ee/workflows/lib/reminders/smsProviders/twilioProvider.ts [hunk 5]
```diff
@@ -39,7 +39,7 @@ export const scheduleSMS = async (phoneNumber: string, body: string, scheduledDa
     to: phoneNumber,
     scheduleType: "fixed",
     sendAt: scheduledDate,
-    from: sender,
+    from: sender ? sender : "+17792527404",
```
Comment (baileypumfleet) [inline packages/features/ee/workflows/lib/reminders/smsProviders/twilioProvider.ts:42]: Hard coded phone number here too

File: packages/features/ee/workflows/lib/aphanumericSenderIdSupport.ts (hunk 1 above)
Comment (baileypumfleet) [inline packages/features/ee/workflows/lib/aphanumericSenderIdSupport.ts:1]: This file also is missing the 'l' in 'alpha'

File: packages/features/ee/workflows/lib/reminders/smsReminderManager.ts [hunk 6]
```diff
@@ -10,6 +10,7 @@ import dayjs from "@calcom/dayjs";
 import prisma from "@calcom/prisma";
 import { Prisma } from "@calcom/prisma/client";
 
+import { noAlphanumericSenderIdSupport } from "../aphanumericSenderIdSupport";
```
Comment (baileypumfleet) [inline packages/features/ee/workflows/lib/reminders/smsReminderManager.ts:13]: Import will also need to be updated here for the spelling error

**Review by pumfleet — DISMISSED** [review]
Great PR! I've left a couple of comments about a file name (and subsequently it's imports) being misspelt as well as a hardcoded phone number, but otherwise all good :)

File: packages/features/ee/workflows/lib/reminders/smsProviders/twilioProvider.ts [hunk 7]
```diff
@@ -25,7 +25,7 @@ export const sendSMS = async (phoneNumber: string, body: string, sender: string)
     body: body,
     messagingServiceSid: process.env.TWILIO_MESSAGING_SID,
     to: phoneNumber,
-    from: sender,
+    from: sender ? sender : "+17792527404",
```
[the remark this answers is not in the document]
Comment (CarinaWolli) [reply packages/features/ee/workflows/lib/reminders/smsProviders/twilioProvider.ts:28]: Yes, shouldn't be hardcoded. That's our twilio phone number. @zomars Should we add it as a .env variable?

File: packages/features/ee/workflows/lib/aphanumericSenderIdSupport.ts (hunk 1 above)
Comment (CarinaWolli) [reply packages/features/ee/workflows/lib/aphanumericSenderIdSupport.ts:1]: Sorry for the spelling mistakes. Thank you for pointing them out 🙏

**Review by emrysal — CHANGES_REQUESTED** [review]
```javascript
- const isAlphanumericSenderIdSupported = !noAlphanumericSenderIdSupport.find(
-     (code) => code === reminderPhone?.substring(0, code.length)
-   );
-   const senderID = isAlphanumericSenderIdSupported ? sender : "";
+ getSenderId(reminderPhone)
// or
+ getSenderId(sendTo) // returns process.env.TWILIO_SENDER_ID = "Cal" || ""
```

1) `getSenderId` Should be the function exposed through the library - this way you can reduce the duplicate code.
2) The SENDER_ID should be an environment variable (optional) - defaulting to Cal - allowing self hosters to easily update the SENDER_ID. Perhaps should be non-twilio-specific.

File: packages/features/ee/workflows/lib/alphanumericSenderIdSupport.ts [hunk 8]
```diff
@@ -0,0 +1,87 @@
+import { SENDER_ID } from "@calcom/lib/constants";
+
+export function getSenderId(phoneNumber?: string | null, sender?: string | null) {
```
Comment (CarinaWolli) [inline packages/features/ee/workflows/lib/alphanumericSenderIdSupport.ts:3]: @emrysal added the `getSenderId` function and also added a new constant `SENDER_ID` which is the new `NEXT_PUBLIC_SENDER_ID` env variable (if empty, "Cal" as a fallback)

File: packages/features/ee/workflows/lib/aphanumericSenderIdSupport.ts (hunk 1 above)
Comment (PeerRich) [reply packages/features/ee/workflows/lib/aphanumericSenderIdSupport.ts]: do we throw an error if the country is not supported?

**Review by emrysal — APPROVED** [review]
🚀 Perfect!

**Pull request merged**